aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -108,43 +108,62 @@ c10::DispatchKeySet generate_nested_key_set(at::Tensor buffer) {
 }
 
 NestedTensorImpl::NestedTensorImpl(
-    at::Tensor buffer,
+    int64_t buffer_size,
+    Storage storage,
```

Comment:
It has been a long day of l-value vs r-value semantics for me I think this right but honestly little confused. If someone who has a good grasp on all the const l value refs vs pass by value vs pass by r-value stuff and how it applies to our particular class types e.g. Tensor (which is a pointer class to tensor impl) and Storage (which is pointer class to storage impl) I would be eternally grateful and indebted to you for life  if you would vc me and ELI5 lol

Reply:
The rule is pretty simple: if you *always* take ownership of the passed in object, then you should probably take it by value. This is definitely true for smart pointer types. Scott Meyers has a good discussion about this, see Item 41 in https://moodle.ufsc.br/pluginfile.php/2377667/mod_resource/content/0/Effective_Modern_C__.pdf